pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -266,9 +279,37 @@ func (m *AddOnManager) verifyRequiredVariablesInContext(context *command.Executi
 
 	if len(missingVars) > 0 {
 		missing := strings.TrimSpace(strings.Join(missingVars, ", "))
-		return errors.New(fmt.Sprintf("The variable(s) %s are required by the add-on, but are not defined in the context.", missing))
+		return fmt.Errorf("The variable(s) %s are required by the add-on, but are not defined in the context", missing)
+	}
+
+	return nil
+}
+
+func (m *AddOnManager) verifyRequiredOpenshiftVersion(context *command.ExecutionContext, meta addon.AddOnMeta) error {
+	dockerCommander := context.GetDockerCommander()
+	versionInfo, err := dockerCommander.Exec(" ", constants.OpenshiftContainerName, "openshift", "version")
+	if err != nil {
+		return err
 	}
 
+	// verionInfo variable have below string as value along with new line
+	// openshift v3.6.0+c4dd4cf
+	// kubernetes v1.6.1+5115d708d7
+	// etcd 3.2.1
+	// openShiftVersionAlongWithCommitSha is contain *v3.6.0+c4dd4cf* (first split on new line and second on space)
+	openShiftVersionAlongWithCommitSha := strings.Split(strings.Split(versionInfo, "\n")[0], " ")[1]
+	// openshiftVersion is contain *3.6.0* (split on *+* string and then trim the *v* as perfix)
+	// TrimSpace is there to make sure no whitespace around version string
+	openShiftVersion := strings.TrimSpace(strings.TrimPrefix(strings.Split(openShiftVersionAlongWithCommitSha, "+")[0], minikubeConstants.VersionPrefix))
+	requiredOpenshiftVersions := strings.TrimSpace(meta.OpenShiftVersion())
+	if requiredOpenshiftVersions != "" {
+		for _, requiredOpenshiftVersion := range strings.Split(requiredOpenshiftVersions, ",") {
```

Comment:
Maybe we can introduce ',' as constant - VersionRangeSeperator? Just to make the code more explicit

Reply:
@hferentschik Agree, will do 👍